core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java

```diff
@@ -358,7 +358,7 @@ private User createAccount(StaplerRequest req, StaplerResponse rsp, boolean self
         if(si.fullname==null || si.fullname.length()==0)
             si.fullname = si.username;
 
-        if(si.email==null || !si.email.contains("@"))
+        if(isMailerPluginPresent() && (si.email==null || !si.email.contains("@")))
```

Comment:
> This is really crying out for an extension point

Agreed, would be great, but out of scope for this PR and 2.0 I think. 2.1 maybe…